data/scenarios/Challenges/_dimsum/solution.sw

```diff
@@ -0,0 +1,27 @@
+def doN = \n. \f. if (n > 0) {f; doN (n - 1) f} {}; end;
```

Comment:
Notice that the solution uses recursion, but the base robot does not have a `strange loop`.  This should not be allowed but currently we can get away with it because of #231 .  I forgot about this bug and assumed that I was not allowed to use recursion in my solution!  But we should add a `strange loop` and a `lambda` to the `base` inventory.